engine/crates/engine-v2/src/execution/coordinator.rs

```diff
@@ -135,14 +122,19 @@ impl<'ctx> ExecutorCoordinator<'ctx> {
         &'ctx self,
         mut futures: ExecutorFutureSet<'ctx>,
         mut response: ResponseBuilder,
-        mut mutation_root_fields_executors: Vec<Executor<'ctx>>,
         planner: &mut Planner<'ctx>,
+        plans_with_dependencies: Vec<Plan>,
     ) -> Response {
-        while let Some(ExecutorFutResult {
-            result,
-            is_mutation_root_field,
-        }) = futures.next().await
-        {
+        let mut outgoing_edges = HashMap::<PlanId, Vec<PlanId>>::new();
+        let mut plans = HashMap::<PlanId, Plan>::new();
+        for plan in plans_with_dependencies {
+            for dependency in &plan.sibling_dependencies {
+                outgoing_edges.entry(*dependency).or_default().push(plan.id);
+            }
+            plans.insert(plan.id, plan);
+        }
```

Comment:
Feel like there might be some way to wrap this (and your other changes in this function) up so they're easier to follow.

Reply:
I think it did better, not great, but ok for now 🤷

Reply:
Much better, thanks